internal/pkg/controllers/profile/profile.go

```diff
@@ -85,42 +87,66 @@ func isProfile(obj runtime.Object) bool {
 func Setup(mgr ctrl.Manager, l logr.Logger) error {
 	const name = "profile"
 
-	return ctrl.NewControllerManagedBy(mgr).
+	if err := ctrl.NewControllerManagedBy(mgr).
 		Named(name).
 		For(&corev1.ConfigMap{}).
 		WithEventFilter(resource.NewPredicates(isProfile)).
 		Complete(&Reconciler{
 			client: mgr.GetClient(),
 			log:    l,
 			record: event.NewAPIRecorder(mgr.GetEventRecorderFor("profile")),
+			save:   saveProfileOnDisk,
+		}); err != nil {
+		return err
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&seccompoperatorv1alpha1.SeccompProfile{}).
+		Complete(&Reconciler{
+			client: mgr.GetClient(),
+			log:    l,
+			record: event.NewAPIRecorder(mgr.GetEventRecorderFor("profile")),
+			save:   saveProfileOnDisk,
 		})
 }
 
+type saver func(string, []byte) error
+
 // A Reconciler reconciles seccomp profiles.
 type Reconciler struct {
 	client client.Client
 	log    logr.Logger
 	record event.Recorder
+	save   saver
```

Comment:
Is this `saver` still required?

Reply:
I added it to help with the unit tests. Without it, [this new test](https://github.com/kubernetes-sigs/seccomp-operator/pull/125/files#diff-e4e0872e41f6b6c837b6e05708f07bf2R146-R154) tries to write the profile to /var/lib/. I'm new to testing and mocking in Go so open to other suggestions for this.

Reply:
Ah okay, yeah this seems reasonable. Another option would be to use an interface, but this function seems a good leightweight alternative to me. 👍